app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{email.downcase!}
+  validates :name,  presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
Khai báo dòng này mục đích là để làm gì vậy em? cơ chế hoạt động ra làm sao?

Reply:
nó lưu mật khẩu dưới dạng hash thay vì mật khẩu thực,khi log in sẽ so sánh với hash trong csdl